### PR TITLE
Add generator config saving and loading

### PR DESCRIPTION
**Pull request by Piphi5**
Users can now save their config files and load them again in the generator

### CODE REVIEW

File: sysid-application/src/main/native/cpp/view/Generator.cpp [hunk 1]
```diff
@@ -6,13 +6,16 @@
...
+#include <stdexcept>
 #include <string>
+#include <string_view>
 
 #include <fmt/format.h>
 #include <glass/Context.h>
 #include <imgui.h>
 #include <imgui_stdlib.h>
 #include <wpi/Logger.h>
+#include <wpi/span.h>
```
Comment (calcmogul) [inline sysid-application/src/main/native/cpp/view/Generator.cpp:18]: Is this code actually using wpi::span?

File: sysid-application/src/main/native/cpp/generation/ConfigManager.cpp [hunk 2]
```diff
@@ -61,10 +64,94 @@ wpi::json ConfigManager::Generate(size_t occupied) {
...
+  if (ec) {
+    throw std::runtime_error(fmt::format("Unable to read: {}", path));
+  }
+
+  wpi::json json_file;
+  is >> json_file;
+  WPI_INFO(m_logger, "Read {}", path);
+
+  for (auto&& key : json_keys) {
+    if (json_file.find(std::string(key)) == json_file.end()) {
```
Comment (calcmogul) [inline sysid-application/src/main/native/cpp/generation/ConfigManager.cpp:106]: Does `find()` not have a std::string_view overload?

File: sysid-application/src/main/native/cpp/view/Generator.cpp (hunk 1 above)
Comment (Piphi5) [reply sysid-application/src/main/native/cpp/view/Generator.cpp]: No it's not.

File: sysid-application/src/main/native/cpp/generation/ConfigManager.cpp [hunk 3]
```diff
@@ -61,10 +64,94 @@ wpi::json ConfigManager::Generate(size_t occupied) {
...
+  }
+
+  wpi::json json_file;
+  is >> json_file;
+  WPI_INFO(m_logger, "Read {}", path);
+
+  for (auto&& key : json_keys) {
+    if (json_file.find(std::string_view(key)) == json_file.end()) {
+      throw std::runtime_error(fmt::format(
+          "{} was not present in config file. Please check its formatting",
```
Comment (prateekma) [inline sysid-application/src/main/native/cpp/generation/ConfigManager.cpp:108]: Add period at end of sentence.

**Review by prateekma — CHANGES_REQUESTED** [review]

**Review by calcmogul — APPROVED** [review]

**Review by prateekma — APPROVED** [review]

**Pull request merged**